cmd/cake/main.go

```diff
@@ -67,10 +67,9 @@ func main() {
 		}
 	})
 
-	dockerClient := cake.NewExternalDockerClient(config.AuthConfig)
-
 	if !*dryRun {
-		cake.WalkBuildGraph(buildGraph, func(image *cake.Image) {
+		cake.WalkBuildGraphParallel(buildGraph, func(image *cake.Image) {
+			dockerClient := cake.NewExternalDockerClient(config.AuthConfig)
```

Comment:
is it necessary to create `dockerClient` each time the `apply` function is called or can we use the single instance?

Reply:
AFAIK there's no thread-safety guarantees for Docker client and it seems reasonable to have an independent instance per routine. Let me do a test run with shared Docker Client to see if there any issues.

Reply:
using a shared instance of Docker Client works, let's keep it that way.